server/src/test/info.test.ts

```diff
@@ -0,0 +1,113 @@
+import * as assert from 'assert';
+import * as path from 'path';
+import * as fs from 'fs/promises';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { Location, Position, Range } from 'vscode-languageserver';
+
+import { compileAqua } from '../validation';
+import { DocumentInfo } from '../info';
+import { pathToUri, tokenToLocation } from '../utils';
+
+/**
+ * Load document from file, compile it and return info
+ * @param relPath path relative to test folder
+ * @returns document, document info after compilation
+ */
+async function openDocument(relPath: string): Promise<[TextDocument, DocumentInfo]> {
+    const absPath = path.join(__dirname, relPath);
+    const uri = pathToUri(absPath);
+
+    const file = await fs.open(absPath);
+    const content = await file.readFile();
+    await file.close();
+
+    const document = TextDocument.create(uri, 'aqua', 0, content.toString());
+    // Compile without imports
+    const [_, info] = await compileAqua({ imports: {} }, document);
+
+    return [document, info];
+}
+
+/**
+ * Find all locations of a variable in a document
+ * @param name variable name
+ * @param doc text document to search in
+ * @returns locations of variable in document
+ */
+function locationsOf(name: string, doc: TextDocument): Location[] {
+    const re = new RegExp(`(?<=\\b)${name}(?=\\b)`, 'g');
+    const text = doc.getText();
+
+    const locations: Location[] = [];
+    for (let match = re.exec(text); match !== null; match = re.exec(text)) {
+        const start = doc.positionAt(match.index);
+        const end = doc.positionAt(match.index + match[0].length);
+        locations.push({
+            uri: doc.uri,
+            range: {
+                start,
+                end,
+            },
+        });
+    }
+
+    return locations;
+}
```

Comment:
Also something like this is probably a bit simpler if `matchAll` works for you
```suggestion
function locationsOf(name: string, doc: TextDocument): Location[] {
    return [...doc.getText().matchAll(new RegExp(`(?<=\\b)${name}(?=\\b)`, 'g')].map((match) => {
        return {
            uri: doc.uri,
            range: {
                start: doc.positionAt(match.index),
                end: doc.positionAt(match.index + match[0].length),
            },
        }
    })
}
```